pinot-core/src/main/java/org/apache/pinot/core/query/reduce/StreamingReduceService.java

```diff
@@ -127,23 +129,19 @@ private static void processIterativeServerResponse(StreamingReducer reducer, Exe
               aggregator.aggregate(entry.getKey(), dataTable);
             }
           }
+          return null;
```

Comment:
why return null here?

Reply:
+1, using `runAsync` instead would avoid this.

Reply:
Yes good point. I think it was originally something with throwing Exceptions while `ExecutorService.submit` is called, and the Future<Void> asks for a return value (at least in my IDE it is red).
Anyway, changed now.